amazon/ion/simpleion.py

```diff
@@ -138,14 +138,17 @@ def dump(obj, fp, imports=None, binary=True, sequence_as_stream=False, skipkeys=
         iterable_as_array: NOT IMPLEMENTED
         tuple_as_sexp (Optional[True|False]): When True, all tuple values will be written as Ion s-expressions.
             When False, all tuple values will be written as Ion lists. Default: False.
+        omit_version_marker (Optional|True|False): If binary is False and omit_version_marker is True, omits the
+            Ion Version Marker ($ion_1_0) from the output.  Default: False.
         **kw: NOT IMPLEMENTED
 
     """
 
     raw_writer = binary_writer(imports) if binary else text_writer(indent=indent)
     writer = blocking_writer(raw_writer, fp)
     from_type = _FROM_TYPE_TUPLE_AS_SEXP if tuple_as_sexp else _FROM_TYPE
-    writer.send(ION_VERSION_MARKER_EVENT)  # The IVM is emitted automatically in binary; it's optional in text.
+    if not binary and not omit_version_marker:
```

Comment:
Shouldn't this be `if binary or not omit_version_marker:`?

I mean, the tests pass so the current logic clearly works, but it looks weird to me. Are we relying on the fact that under the hood the binary writer always writes the ION_VERSION_MARKER_EVENT without being explicitly told to?

Reply:
Agreed, thanks for catching this.  Fixing in https://github.com/amzn/ion-python/pull/111